### PR TITLE
Allow unauthenticated users to view the courses page

### PR DESCRIPTION
**Pull request by caramelcat**
1473_unauthenticated_user_can_view_courses_page

### CODE REVIEW

File: app/controllers/course/courses_controller.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 class Course::CoursesController < Course::Controller
   include Course::ActivityFeedsConcern
 
+  skip_before_action :authenticate_user!, only: :index
```
Comment (fonglh) [inline app/controllers/course/courses_controller.rb]: `ApplicationUserConcern` has some way to skip `authenticate_user!` in https://github.com/Coursemology/coursemology2/blob/master/app/controllers/concerns/application_user_concern.rb#L11, depending on `publicly_accessible?`

I'm not sure, but maybe it's possible to make use of this to skip the auth check by overriding `publicly_accessible?` somewhere?

Comment (caramelcat) [inline app/controllers/course/courses_controller.rb]: @fonglh I've made the changes, as suggested.

Comment (fonglh) [inline app/controllers/course/courses_controller.rb]: LGTM! will let @allenwq double check and merge.

**Comment by weiqingtoh** [comment]
Possible to put it as a protected method instead?

**Comment by caramelcat** [comment]
@weiqingtoh ok, I made the changes, as suggested.

**Comment by weiqingtoh** [comment]
LGTM!

**Pull request merged**